.github/workflows/release.yml

```diff
@@ -120,22 +120,19 @@ jobs:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
           DRY_RUN: true
 
-      - name: "3. Determine docker tags"
-        run: echo ::set-env name=DOCKER_TAGS::$(echo latest, ${{ steps.bump_dry.outputs.tag }})
+      - name: "3. Login to docker hub"
+        run: "echo '${{ secrets.DOCKER_HUB_TOKEN }}' | docker login --username ${{ secrets.DOCKER_HUB_USER }} --password-stdin"
 
-      - name: "3-1. Determine docker tags for hotfix"
+      - name: "4-1. Publish hotfix docker image"
         if: contains(github.event.head_commit.message, '#hotfix')
-        run: echo ::set-env name=DOCKER_TAGS::$(echo ${{ steps.bump_dry.outputs.tag }})
+        run: |
+          ./gradlew dockerTagsPush -P version=${{ steps.bump_dry.outputs.tag }}
+
+      - name: "4-2. Publish non-hotfix docker image"
+        if: ${{ !contains(github.event.head_commit.message, '#hotfix') }}
+        run: |
+          ./gradlew dockerPush -P version=${{ steps.bump_dry.outputs.tag }}
```

Comment:
What's the difference between `dockerPush` and `dockerTagsPush`? It is not intuitive what each method does

Reply:
> What's the difference between `dockerPush` and `dockerTagsPush`? It is not intuitive what each method does

As you can see above in gradle.build file in docker plugin we specify only tag ```version```, but I've mentioned before that gradle automatically creates also tag 'latest'. `dockerPush` pushes image with all **created** tags, wheareas `dockerTagsPush` publish image only with **specified** tags, so `dockerTagsPush` will push image only with one tag - ```version```, while `dockerPush` will push image with both ```version``` and 'latest' tags. I've disscussed value of ```version``` variable in previous comments.

Reply:
can we use always dockerTagsPush? to make it more explicit?

Reply:
> can we use always dockerTagsPush? to make it more explicit?

We can create two tags, one for `latest` and one for current `version` and use a `dockerPush${version}` and optionally `dockerPushLatest`  (See [here](https://github.com/unit8co/darts/pull/159#discussion_r459334192)). In such case only thing that we should agree on is how to pass `version` variable to gradle. I see two possibilities:
1. Pass `version` to gradle as property from CLI in workflow (current implementation)
2. Set `version` as enviromental variable in workflow and use `System.env.version` in gradle to retrieve it. 

What do you think?

Reply:
I would recommend passing it as property. Reading something from env in gradle is not visible at first and might confuse people